src/main/java/com/diffmin/patch/Generation.java

```diff
@@ -0,0 +1,102 @@
+package com.diffmin.patch;
+
+import com.diffmin.SpoonMapping;
+import com.diffmin.util.Pair;
+import com.github.gumtreediff.actions.model.Delete;
+import com.github.gumtreediff.actions.model.Insert;
+import com.github.gumtreediff.actions.model.Update;
+import gumtree.spoon.diff.Diff;
+import gumtree.spoon.diff.operations.Operation;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.IntStream;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.code.CtStatementList;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtExecutable;
+
+/** Class for generating patches. */
+public class Generation {
```

Comment:
Same comment as for `Application`, I think this class would be better named something like `PatchGenerator`.